# Portafolios de Inversión
## Actividad en Clase 3
José Armando Melchor Soto

---

### Activos alternativos

- Los activos alternativos son los activos financieros que no pertenecen a los activos financieros tradicionales de inversión, los bonos (renta fija) y las acciones (renta variable).
- Muchos de los activos alternativos tienen la ventaja de que la mayoría son tangibles. Es decir, que al invertir en bienes raíces, materias primas y obras de arte, se tiene propiedad de algo físico con un valor intrínseco.

    - Ejemplo:
        - Bienes raíces
        - Fondos de capital privado
        - Inversión en materias primas
        - Inversión en obras de arte
        - Criptomonedas
        - Energía renovable
        - Préstamos privados y crowdfunding
        - Artículos coleccionables


---

### Librerías 

In [46]:
import pandas as pd
import numpy as np
import yfinance as yf
import matplotlib.pyplot as plt


---


### Importar datos

In [47]:
prices = yf.download(['^GSPC', 'ACWI',  '^DJI', 'IYF', 'XLY','XLP','^NBI', 'GSG','BND', 'IHE'], start='2020-01-01', end= '2025-06-02')['Close']
prices.head()

[*********************100%***********************]  10 of 10 completed


Ticker           ACWI        BND        GSG        IHE        IYF        XLP  \
Date                                                                           
2020-01-02  73.371559  72.023140  16.290001  49.126122  63.359661  54.455326   
2020-01-03  72.739288  72.280434  16.510000  48.599964  63.002880  54.368168   
2020-01-06  72.904228  72.194679  16.510000  48.658073  62.966316  54.481480   
2020-01-07  72.720963  72.134605  16.459999  48.526535  62.540951  54.063122   
2020-01-08  73.078323  72.023140  16.150000  48.431705  62.925148  54.263577   

Ticker             XLY          ^DJI        ^GSPC         ^NBI  
Date                                                            
2020-01-02  121.261292  28868.800781  3257.850098  3763.050049  
2020-01-03  120.229355  28634.880859  3234.850098  3715.300049  
2020-01-06  120.563789  28703.380859  3246.280029  3743.429932  
2020-01-07  120.372688  28583.679688  3237.179932  3745.149902  
2020-01-08  120.735786  28745.089844  3253.050049  3778.810059

---

### Rendimientos

In [48]:
returns= prices.pct_change().dropna()
returns

Ticker          ACWI       BND       GSG       IHE       IYF       XLP  \
Date                                                                     
2020-01-03 -0.008617  0.003572  0.013505 -0.010710 -0.005631 -0.001601   
2020-01-06  0.002268 -0.001186  0.000000  0.001196 -0.000580  0.002084   
2020-01-07 -0.002514 -0.000832 -0.003029 -0.002703 -0.006755 -0.007679   
2020-01-08  0.004914 -0.001545 -0.018834 -0.001954  0.006143  0.003708   
2020-01-09  0.005643  0.001191 -0.007430  0.004674  0.006542  0.007067   
...              ...       ...       ...       ...       ...       ...   
2025-05-23 -0.003097  0.001526  0.007948  0.001733 -0.001658  0.003690   
2025-05-27  0.016762  0.004710 -0.006957  0.012107  0.017745  0.009069   
2025-05-28 -0.007157 -0.002206 -0.003737 -0.005903 -0.008331 -0.006801   
2025-05-29  0.003645  0.003593 -0.003282  0.010001  0.005110  0.004035   
2025-05-30 -0.002098  0.001928 -0.004233  0.007117  0.000000  0.008769   

Ticker           XLY      ^DJI     ^GSPC      ^NBI  
Date                                                
2020-01-03 -0.008510 -0.008103 -0.007060 -0.012689  
2020-01-06  0.002782  0.002392  0.003533  0.007571  
2020-01-07 -0.001585 -0.004170 -0.002803  0.000459  
2020-01-08  0.003016  0.005647  0.004902  0.008988  
2020-01-09  0.005302  0.007369  0.006655  0.005541  
...              ...       ...       ...       ...  
2025-05-23 -0.008986 -0.006116 -0.006708  0.000747  
2025-05-27  0.029528  0.017801  0.020459  0.009271  
2025-05-28 -0.009176 -0.005785 -0.005571 -0.008829  
2025-05-29  0.000419  0.002780  0.004011  0.020745  
2025-05-30 -0.005443  0.001287 -0.000081 -0.022024  

[1359 rows x 10 columns]

---

### Tasa libre de riesgo

Tomado de la página de Estados Unidos

In [49]:
rf = 4.04

---

In [ ]:
df = pd.DataFrame(columns= returns.keys()).T
df['Rend_Prom_Anual']= returns.mean() * 252 *100
df['Vol_Prom_Anual']=  np.sqrt(252)* returns.std() *100

df['Sharpe_Ratio'] = (df['Rend_Prom_Anual'] - rf) / df['Vol_Prom_Anual']
df['Risk_aversion']= (df['Rend_Prom_Anual'] - rf) / (0.5*df['Vol_Prom_Anual']**2)

df.sort_values(by='Risk_aversion', ascending=False, inplace=True)
df

Rend_Prom_Anual  Vol_Prom_Anual  Sharpe_Ratio  Risk_aversion
Ticker                                                              
^GSPC         13.424816       21.753694      0.431413       0.039663
ACWI          11.755718       20.324032      0.379635       0.037358
XLP            9.216734       16.951444      0.305386       0.036031
IYF           14.471675       25.434305      0.410142       0.032251
XLY           14.051931       26.536499      0.377289       0.028435
^DJI           9.272181       20.932028      0.249961       0.023883
IHE            7.016616       18.953357      0.157050       0.016572
GSG            7.890947       24.487833      0.157260       0.012844
^NBI           4.639452       25.059708      0.023921       0.001909
BND            0.368948        6.905672     -0.531600      -0.153960

Por cada unidad de riesgo obtenido , yo obtendré 0.43 unidades de rendimiento.